src/plugins/controls/public/controls/data_controls/range_slider/get_range_slider_control_factory.tsx

```diff
@@ -81,6 +88,14 @@ export const getRangesliderControlFactory = (): DataControlFactory<
         initialState,
         dataControl.setters.onSelectionChange
       );
+      const hasRangeSliderSelections$ = new BehaviorSubject<boolean>(
+        Boolean(selections.value$.getValue())
+      );
+      const hasSelectionsSubscription = selections.value$
+        .pipe(distinctUntilChanged())
+        .subscribe((hasSelections) => {
+          hasRangeSliderSelections$.next(Boolean(hasSelections));
+        });
```

Comment:
I refactored other places purely for consistency, but can move it back. Should I? I Will do as you suggest 😊

Reply:
I definitely understand wanting to keep things consistent! But in this case, I'm not sure if that consistency is worth the abstraction of `hasRangeSliderSelections$` - let's see what @nreese thinks :)

Reply:
> Since range slider only has one way that selections can be set, I think it might actually be cleaner to just call hasRangeSliderSelections$.next in the setValue method like we were before

I agree with Hannah. We should favor simplicity before consistency.

Reply:
To add more context.

Consistency is important from an outside perspective. In this case, range slider and options list both expose `hasSelections$`. This makes notification of having selections consistent.

There should not be an emphasis on consistency in in the internal implementation details within range slider and options list.